### PR TITLE
Editable list fixes

### PR DESCRIPTION
**Pull request by aab-odoo**
Assume an x2many editable list where the user edits a record (input
field), and then clicks on 'Add a line' (and keeps it pressed for a
while, before releasing the click).

When the 'mousedown' event occurs, the 'change' event is triggered
on the edited field, and (once potential onchanges have been
applied), the list is updated, i.e. all rows except for the one
being edited a re-rendered.

However, the 'Add a line' handler listens to the 'click' event,
which never occurs if we release the click after the rows have been
re-rendered.

This is an issue for every rows, not only 'Add a line', but for
data rows, the issue is already present in former versions, and is
not that easy to fix). The 'Add a line' case is a consequence of
the recent work on list views, so we can fix it independently.

Task 1915702

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo rebase-ff

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ ListRenderer.include({
      * This is a specialized version of confirmChange, meant to be called when
```
Comment (mge-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: point in commit message the commit that broke this "The 'Add a line' case is a consequence of
the recent work on list views"

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 2]
```diff
@@ -444,28 +444,16 @@ var ListRenderer = BasicRenderer.extend({
...
-                    record: list.groupData,
-                });
-            } else {
-                self.trigger_up('button_clicked', {
-                    attrs: node.attrs,
-                    record: list.groupData,
-                });
-            }
-        });
+        $button.on("click", this._onGroupButtonClicked.bind(this, list.groupData, node));
```
Comment (mge-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js]: what about detach in AM ?

Comment (mge-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js]: but TODO, we meant moving it to an entry in `events` key

**Comment by mge-odoo** [comment]
robodoo r+

File: addons/web/static/src/js/widgets/date_picker.js [hunk 3]
```diff
@@ -246,9 +246,22 @@ var DateWidget = Widget.extend({
...
-            this.$el.datetimepicker('hide');
-            this.__libInput--;
+            // tempusdominus doesn't offer any elegant way to check whether the
+            // datepicker is open or not ; however, as we are in a keydown event
+            // handler, we know that this widget has the focus, and thus that no
+            // other datepicker than this one could be opened right now (as it
+            // would have been closed on focusout), so we check for the presence
+            // of a datepicker widget in the DOM to determine whether this one
+            // is open or not
+            if ($('body > .bootstrap-datetimepicker-widget').length) {
```
Comment (mge-odoo) [inline addons/web/static/src/js/widgets/date_picker.js]: as discussed, maybe keep internal state by listen on `hide` and `show`

**Comment by mge-odoo** [comment]
robodoo r+

**Pull request closed without merging**